packages/@aws-cdk/app-staging-synthesizer-alpha/README.md

```diff
@@ -0,0 +1,394 @@
+const resourceApp = new App();
+const resources = new CustomStagingStack(resourceApp, 'CustomStagingStack', {});
+
+const app = new App({
+  defaultStackSynthesizer: AppStagingSynthesizer.customResources({
+    resources,
+  }),
+});
+```
+
+## Default Staging Stack
+
+The Default Staging Stack includes all the staging resources necessary for CDK Assets. The below example
+is of a CDK App using the `AppStagingSynthesizer` and creating a file asset for the Lambda Function
+source code. As part of the `DefaultStagingStack`, an S3 bucket and IAM role will be created that will be
+used to upload the asset to S3.
+
+```ts
+const app = new App({
+  defaultStackSynthesizer: AppStagingSynthesizer.defaultResources({ appId: 'my-app-id' }),
+});
+
+const stack = new Stack(app, 'my-stack');
+
+new lambda.Function(stack, 'lambda', {
+  code: lambda.AssetCode.fromAsset(path.join(__dirname, 'assets')),
+  handler: 'index.handler',
+  runtime: lambda.Runtime.PYTHON_3_9,
+});
+
+app.synth();
+```
+
+### Custom Roles
+
+You can customize some or all of the roles you'd like to use in the synthesizer as well,
+if all you need is to supply custom roles (and not change anything else in the `DefaultStagingStack`):
+
+```ts
+const app = new App({
+  defaultStackSynthesizer: AppStagingSynthesizer.defaultResources({
+    appId: 'my-app-id',
+    deploymentRoles: DeploymentIdentities.specifyRoles({
+      cloudFormationExecutionRole: BootstrapRole.fromRoleArn('arn:aws:iam::123456789012:role/Execute'),
+      deploymentRole: BootstrapRole.fromRoleArn('arn:aws:iam::123456789012:role/Deploy'),
+      lookupRole: BootstrapRole.fromRoleArn('arn:aws:iam::123456789012:role/Lookup'),
+    }),
+  }),
+});
+```
+
+Or, you can ask to use the CLI credentials that exist at deploy-time.
+These credentials must have the ability to perform CloudFormation calls,
+lookup resources in your account, and perform CloudFormation deployment.
+For a full list of what is necessary, see `LookupRole`, `DeploymentActionRole`,
+and `CloudFormationExecutionRole` in the
+[bootstrap template](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk/lib/api/bootstrap/bootstrap-template.yaml).
+
+```ts
+const app = new App({
+  defaultStackSynthesizer: AppStagingSynthesizer.defaultResources({
+    appId: 'my-app-id',
+    deploymentRoles: DeploymentIdentities.cliCredentials(),
+  }),
+});
+```
+
+The default staging stack will create roles to publish to the S3 bucket and ECR repositories,
+assumable by the deployment role. You can also specify an existing IAM role for the
+`fileAssetPublishingRole` or `imageAssetPublishingRole`:
+
+```ts
+const app = new App({
+  defaultStackSynthesizer: AppStagingSynthesizer.defaultResources({
+    appId: 'my-app-id',
+    fileAssetPublishingRole: BootstrapRole.fromRoleArn('arn:aws:iam::123456789012:role/S3Access'), 
+    imageAssetPublishingRole: BootstrapRole.fromRoleArn('arn:aws:iam::123456789012:role/ECRAccess'),
+  }),
+});
+```
+
+### Ephemeral S3 Assets
```

Comment:
I still prefer the term "handoff", no? Do you disagree?

Reply:
discussed and will be using the term `deployTime` instead